### ISSUE TITLE
Can't recieve call on 5G with volte enabled

### ISSUE THREAD

**Issue by weaponmasterjax** [phhusson/treble_experimentations#2473]
On 5g device(Redmi note 11 5g)(evergo),the ims app does work with ordinary LTE and enables volte.
But there is no settings for VoNR and SA networks.
AND NOT RECIEVING INCOMING CALLS.
This is on a MTK device.
Custom roms with mtk telephony provider seems to work fine.

**Comment by 8y386854** [comment]
I have the same issue. I cannot receive calls after turning on Volte. However, there is no problem regarding calling out and receive SMS. My Realme V3 5g. It is a Mediatek mobile. Is there a solution?

**Comment by weaponmasterjax** [comment]
> I have the same issue. I cannot receive calls after turning on Volte. However, there is no problem regarding calling out and receiving SMS. My mobile is Realme V3 5g, it is a Mediatek one. Is there a solution?

yes. Because memetek used a different framework implementation for 5g ims so the default AOSP one wont work.
So the best solution is to find specific aosp rom for your device.

**Comment by phhusson** [comment]
Are you saying there is a fix in device specific rom?
If yes please provide a link to that fix

**Comment by weaponmasterjax** [comment]
> 

https://github.com/dm700-devs/device_xiaomi_camellia/commit/8e027f081420fd8d149bb5708466ed2738101d91

and probably also
https://github.com/dm700-devs/device_xiaomi_camellia/commit/41e67134d2cbf8a10f82d6f854e6044465643a3c
and
https://github.com/dm700-devs/device_xiaomi_camellia/commit/ac29d44b2ffce2eaa189db5a693db29762738b19

**Comment by phhusson** [comment]
All those three are already embedded in gsi

**Comment by 8y386854** [comment]
> > I have the same issue. I cannot receive calls after turning on Volte. However, there is no problem regarding calling out and receiving SMS. My mobile is Realme V3 5g, it is a Mediatek one. Is there a solution?
> 
> yes. Because memetek used a different framework implementation for 5g ims so the default AOSP one wont work. So the best solution is to find specific aosp rom for your device.

Sadly, I can't find a specific aosp rom or custom rom for my Realme V3. I have tried almost all a12 and a13 gsi roms, but got the same results.
